src/decoder/plugins/MadDecoderPlugin.cxx

```diff
@@ -517,8 +517,7 @@ parse_xing(struct xing *xing, struct mad_bitptr *ptr, int *oldbitlen) noexcept
 	if (xing->flags & XING_TOC) {
 		if (bitlen < 800)
 			return false;
-		for (unsigned char & i : xing->toc)
-			i = mad_bit_read(ptr, 8);
+		std::fill(xing->toc.begin(), xing->toc.end(), mad_bit_read(ptr, 8));
```

Comment:
This looks very broken. This reduces 100 `mad_bit_read()` calls to just one, by moving it out of the loop.

Reply:
Hrm good point.